platform/linux-generic/odp_ipsec.c

```diff
@@ -876,6 +956,8 @@ static ipsec_sa_t *ipsec_in_single(odp_packet_t pkt,
 		goto post_lifetime_err_cnt_update;
 	}
 
+	state.in.trl_len += ipsec_get_seqh_len(ipsec_sa);
```

Comment:
> This corrupts the IP length field of the received packets with individual algorithms. Apparently the patch has not been tested.

All the ESN and non-ESN tests are passed with this change.  I will if something is missing there.

 

> but a couple of lines later it is also used for subtracting the trimmed length from the IP packet length. And since the seq_hi field was not present in the actual packet received from the network, it is not included in the IP length and thus should not be subtracted from it.

you meant this line 
`state.ip_tot_len -= state.in.trl_len;`

Reply:
Agreed, its correct that the ESN higher bytes are just has to be truncated from the tail, not supposed to be subtracted from the IP length, changes done as below:
1) Removed this line

> state.in.trl_len += ipsec_get_seqh_len(ipsec_sa);

2) Added the ESN higher bits part in "odp_packet_trunc_tail( )".

> odp_packet_trunc_tail(&pkt, state.in.trl_len + ipsec_get_seqh_len(ipsec_sa), NULL, NULL)